workflow/usage.rst

```diff
@@ -198,6 +211,14 @@ events are dispatched:
 * ``workflow.[workflow name].announce``
 * ``workflow.[workflow name].announce.[transition name]``
 
+When a state transition is initiated, the events are fired in the following order:
+
+- guard: Validate whether the transition is allowed at all (see below);
+- leave: The object is about to leave a place;
+- transition: The object is going through this transition;
+- enter: The object entered a new place. This is the first event where the object' is marked as being in the new place;
+- announce: Triggered once for each workflow that now is available for the object.
```

Comment:
I wonder if we could integrate this with the event list above so it's all in one place. Perhaps the name (guard), then indented below the description, then bullets below that with the 3 events. We do something here https://github.com/symfony/symfony-docs/blob/master/introduction/http_fundamentals.rst for GET POST stuff